### PR TITLE
Updated RCCL introduction at docs/source/library.rst

### PR DESCRIPTION
**Pull request by PedramAlizadeh**

### CODE REVIEW

**Review by wenkaidu — APPROVED** [review]

File: docs/source/library.rst [hunk 1]
```diff
@@ -10,4 +10,7 @@ RCCL
 Introduction
 ------------
 
-The RCCL is an AMD port of NCCL.
+The ROCm Collective Communication Library (RCCL) is a stand-alone library which provides multi-GPU and multi-node collective communication primitives optimized for AMD GPUs and can be easily integrated into applications. 
+RCCL (pronounced “Rickel”) implements routines such as all-reduce, all-gather, reduce, broadcast, reduce-scatter, gather, scatter, all-to-allv, and all-to-all as well as direct point-to-point (GPU-to-GPU) send and receive operations. The provided collective communication routines are implemented using Ring and Tree algorithms and are optimized to achieve high bandwidth and low latency over PCIe and xGMI high-speed interconnects for intra-node communication and over InfiniBand, RoCE, and TCP/IP for inter-node communication. 
+RCCL supports an arbitrary number of GPUs installed in a single-node or multi-node platform and can be used in either single- or multi-process (e.g., MPI) applications.
```
Comment (saadrahim) [inline docs/source/library.rst:15]: Do we need to mention RDMA, HSA?

File: docs/source/library.rst [hunk 2]
```diff
@@ -10,4 +10,7 @@ RCCL
 Introduction
 ------------
 
-The RCCL is an AMD port of NCCL.
+The ROCm Collective Communication Library (RCCL) is a stand-alone library which provides multi-GPU and multi-node collective communication primitives optimized for AMD GPUs and can be easily integrated into applications.
```
Comment (saadrahim) [inline docs/source/library.rst:13]: Please write add a separate sentence for 
and can be easily integrated into applications.

File: docs/source/library.rst [hunk 3]
```diff
@@ -10,4 +10,7 @@ RCCL
 Introduction
 ------------
 
-The RCCL is an AMD port of NCCL.
+The ROCm Collective Communication Library (RCCL) is a stand-alone library which provides multi-GPU and multi-node collective communication primitives optimized for AMD GPUs and can be easily integrated into applications. 
+RCCL (pronounced “Rickel”) implements routines such as all-reduce, all-gather, reduce, broadcast, reduce-scatter, gather, scatter, all-to-allv, and all-to-all as well as direct point-to-point (GPU-to-GPU) send and receive operations. The provided collective communication routines are implemented using Ring and Tree algorithms and are optimized to achieve high bandwidth and low latency over PCIe and xGMI high-speed interconnects for intra-node communication and over InfiniBand, RoCE, and TCP/IP for inter-node communication.
```
Comment (saadrahim) [inline docs/source/library.rst:14]: Too many ands, please separate into multiple sentences.

**Comment by saadrahim** [comment]
@PedramAlizadeh please make a new PR to address the feedback.

**Pull request merged**